src/ModalCard.php

```diff
@@ -0,0 +1,375 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Bulma;
+
+use InvalidArgumentException;
+use JsonException;
+use Yiisoft\Html\Html;
+
+/**
+ * ModalCard renders a modal window that can be toggled by clicking on a button.
+ *
+ * The following example will show the content enclosed between the {@see begin()} and {@see end()} calls within the
+ * modal window:
+ *
+ * ```php
+ * ModalCard::begin()->start();
+ *
+ * echo 'Say hello...';
+ *
+ * echo ModalCard::end();
+ * ```
+ *
+ * @link https://bulma.io/documentation/components/modal/
+ */
+final class ModalCard extends Widget
+{
+    public const SIZE_SMALL = 'is-small';
+    public const SIZE_MEDIUM = 'is-medium';
+    public const SIZE_LARGE = 'is-large';
+    private const SIZE_ALL = [
+        self::SIZE_SMALL,
+        self::SIZE_MEDIUM,
+        self::SIZE_LARGE,
+    ];
+
+    public const COLOR_PRIMARY = 'is-primary';
+    public const COLOR_LINK = 'is-link';
+    public const COLOR_INFO = 'is-info';
+    public const COLOR_SUCCESS = 'is-success';
+    public const COLOR_WARNING = 'is-warning';
+    public const COLOR_DANGER = 'is-danger';
+    private const COLOR_ALL = [
+        self::COLOR_PRIMARY,
+        self::COLOR_LINK,
+        self::COLOR_INFO,
+        self::COLOR_SUCCESS,
+        self::COLOR_WARNING,
+        self::COLOR_DANGER,
+    ];
+
+    private array $options = [];
+    private array $closeButtonOptions = [];
+    private string $closeButtonSize = '';
+    private bool $closeButtonEnabled = true;
+    private string $toggleButtonLabel = 'Toggle button';
+    private string $toggleButtonSize = '';
+    private string $toggleButtonColor = '';
+    private array $toggleButtonOptions = [];
+    private bool $toggleButtonEnabled = true;
+    private string $title = '';
+    private array $titleOptions = [];
+    private string $footer = '';
+    private array $footerOptions = [];
+    private array $headerOptions = [];
+
+    public function start(): string
+    {
+        $this->buildOptions();
+
+        $html = '';
+        $html .= $this->renderToggleButton() . "\n";
+        $html .= Html::beginTag('div', $this->options) . "\n"; // .modal
+        $html .= Html::tag('div', '', ['class' => 'modal-background']) . "\n";
+        $html .= Html::beginTag('div', ['class' => 'modal-card']) . "\n"; // .modal-card
+        $html .= $this->renderHeader();
+        $html .= $this->renderBodyBegin() . "\n";
+
+        return $html;
+    }
+
+    protected function run(): string
+    {
+        $html = '';
+        $html .= $this->renderBodyEnd() . "\n";
+        $html .= $this->renderFooter() . "\n";
```

Comment:
@samdark

With this arrangement of the method `renderFooter ` adding options becomes impossible. Please tell me how to solve this issue?

```php
echo Modal::begin()
    ->footerOptions(['class' => 'bg-transparent'])
    ->start();
...
```

Reply:
Doesn't the code above work well?

Reply:
> Doesn't the code above work well?

Yes, please take a look: https://github.com/yiisoft/yii-bulma/pull/20/files#diff-3a5fcb7d5f593e26c99e016ab2c8513bf9283d200e23ae60b9f6b11fde03e200R38

Reply:
Seems it's a bug in base widget: https://github.com/yiisoft/widget/pull/21